src/fsharp/vs/ServiceDeclarations.fs

```diff
@@ -91,6 +72,23 @@ type FSharpToolTipText = FSharpToolTipText<string>
 // specialization that stores data as <see cref="Layout"/>
 type internal FSharpStructuredToolTipText = FSharpToolTipText<Layout>
 
+/// Interface that defines methods for comparing objects using partial equality relation
+type IPartialEqualityComparer<'T> = 
+    inherit IEqualityComparer<'T>
+    /// Can the specified object be tested for equality?
+    abstract InEqualityRelation : 'T -> bool
+
+module EnvMisc2 =
```

Comment:
Diff minimization: Was it necessary to move this chunk? If not please move it back? thanks